### PR TITLE
[DOC] add aptos wallet adapter docs

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Adds documentation for Aptos Wallet Adapter under the "BUILD APPS" section

### CODE REVIEW

**Review by clay-aptos — COMMENTED** [review]
This looks great, Maayan. Please give me until tomorrow EOD to review.

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+import ThemedImage from '@theme/ThemedImage';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Wallet Adapter Concept
+
+Aptos provides a [monorepo](https://github.com/aptos-labs/aptos-wallet-adapter) “Aptos Wallet Adapter” for wallet and dapps creators for easy development and smooth process in building projects on the Aptos network.
+
+The Aptos Wallet Adapter acts as a service between dapps and wallets and exposes APIs for dapps to interact with the wallets by following our [Wallet Standard](../guides/wallet-standard).
```
Comment (0xmaayan) [inline developer-docs-site/docs/concepts/wallet-adapter-concept.md:13]: @clay-aptos it relies on the change you had in https://github.com/aptos-labs/aptos-core/pull/5806

Comment (clay-aptos) [reply developer-docs-site/docs/concepts/wallet-adapter-concept.md:13]: Ack. Thanks!

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   - Lets them be self-service and support themselves without too much friction.
+   - Prevents build failures in case of any bugs/bad implementation/wrong config files/etc.
+* Simplicity - keeps the Provider package very light and small as the major logic is implemented in the core package.
+* Flexibility - for wallets in creating and implementing custom functions.
+
+<center>
+<ThemedImage
+alt="Wallet Adapter Concept"
+sources={{
+    light: useBaseUrl('/img/docs/10-adapter-chart-light.svg'),
```
Comment (clay-aptos) [inline developer-docs-site/docs/concepts/wallet-adapter-concept.md:71]: Is there significance in the `10-` prefix of images? Can we remove it? I know Raj used numbers when more than one image exists on a page.

File: developer-docs-site/docs/guides/wallet-adapter-for-dapp.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Install wallet dependencies you want to include in your app. You can find a list of the wallets [here](https://github.com/aptos-labs/aptos-wallet-adapter#supported-wallet-packages)
+
+Install the react provider
+
+```bash
+npm install @aptos-labs/wallet-adapter-react
+```
+
+Import dependencies.
```
Comment (clay-aptos) [inline developer-docs-site/docs/guides/wallet-adapter-for-dapp.md:37]: How import dependencies?

File: developer-docs-site/docs/guides/wallet-adapter-for-dapp.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+On the App.jsx file, Import the installed wallets.
+
+```js
+import { AptosWallet } from "some-aptos-wallet-package";
```
Comment (clay-aptos) [inline developer-docs-site/docs/guides/wallet-adapter-for-dapp.md:41]: Can we give them an actual example to accompany `some-aptos-wallet-package` here

**Review by clay-aptos — CHANGES_REQUESTED** [review]
Great work, @0xmaayan ! I made some edits directly to the files. And I included one TODO asking for how to import dependencies. Once that is resolved, this LGTM. Thanks again!

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md (hunk 1 above)
Comment (clay-aptos) [reply developer-docs-site/docs/concepts/wallet-adapter-concept.md:13]: Merged.

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md (hunk 2 above)
Comment (0xmaayan) [reply developer-docs-site/docs/concepts/wallet-adapter-concept.md]: not really, just following a convention I thought we have, fixed it

File: developer-docs-site/docs/guides/wallet-adapter-for-dapp.md (hunk 3 above)
Comment (0xmaayan) [reply developer-docs-site/docs/guides/wallet-adapter-for-dapp.md]: maybe a title should fit better here, fixed it

File: developer-docs-site/docs/guides/wallet-adapter-for-dapp.md (hunk 4 above)
Comment (0xmaayan) [reply developer-docs-site/docs/guides/wallet-adapter-for-dapp.md]: done

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+title: "Aptos Wallet Adapter"
+id: "wallet-adapter-concept"
+---
+
+import ThemedImage from '@theme/ThemedImage';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Aptos Wallet Adapter
+
+Aptos provides an [Aptos Wallet Adapter](https://github.com/aptos-labs/aptos-wallet-adapter) monorepo for wallet and dapps creators to ease development and ensure a smooth process in building projects on the Aptos network.
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/wallet-adapter-concept.md:11]:
```suggestion
We provide an [Aptos Wallet Adapter](https://github.com/aptos-labs/aptos-wallet-adapter) for wallet and dapps creators to ease development and ensure a smooth process in building projects on the Aptos network.
```

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+The Aptos Wallet Adapter acts as a service between dapps and wallets and exposes APIs for dapps to interact with the wallets by following our [Wallet Standard](../guides/wallet-standard).
```
Comment (davidiw) [inline developer-docs-site/docs/concepts/wallet-adapter-concept.md:13]:
```suggestion
The Aptos Wallet Adapter acts as a service between dapps and wallets and exposes APIs for dapps to interact with wallets following our [Wallet Standard](../guides/wallet-standard). This in turns allows dapps to support many wallets with minimal integration efforts.
```

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md (hunk 5 above)
Comment (davidiw) [inline developer-docs-site/docs/concepts/wallet-adapter-concept.md:11]: these paragraphs don't really describe the problem that the wallet adapter is addressing...

1. There are a lot of wallets in the ecosystem
2. Supporting each one is a burden but not supporting them is risky for projects
3. Integration with wallets can be a complex process that requires a lot of testing

A wallet adapter solves these pain points....

or something

Comment (0xmaayan) [reply developer-docs-site/docs/concepts/wallet-adapter-concept.md:11]: we explain it here in details, https://github.com/aptos-labs/aptos-core/pull/5812/files#diff-28a8dc8c00ae95b5f3d12241e42fbdcaf8e022df85c141d748d8de3cb2113001R8-R23

but we can also add some of that on that page! maybe benefits for both wallets and dapps

**Comment by 0xmaayan** [comment]
@clay-aptos please re-review the changes on https://github.com/aptos-labs/aptos-core/pull/5812/files#diff-bc6f3c4dd609b1b3fdf0dffba82abeb7e30e4d788a37b8f2370430171f5287fe

File: developer-docs-site/docs/concepts/wallet-adapter-concept.md (hunk 2 above)
Comment (clay-aptos) [reply developer-docs-site/docs/concepts/wallet-adapter-concept.md]: Perfect. Thank you!

**Review by clay-aptos — APPROVED** [review]
Thank you, Maayan! This LGTM with some small changes I made directly. Thanks again!

**Pull request merged**